Solution: Cyren Threat Intelligence (Official)

mazamizo21 opened the pull request:
## Official Data443 Submission

This is the official submission from the Data443 organization for the **Cyren Threat Intelligence** solution.

### Solution Overview
- **Publisher**: Data443 Risk Mitigation, Inc.
- **Type**: CCF (Codeless Connector Framework) Data Connector
- **ARM-TTK**: 49/50 pass (1 known false positive - IDs Should Be Derived From ResourceIDs)

### Components
- CCF Data Connector for IP Reputation and Malware URLs feeds
- Custom Log Analytics table (Cyren_Indicators_CL)
- Data Collection Rules with KQL transformations
- Analytics Rules for high-risk indicator detection
- Workbooks for threat intelligence visualization

### Validation
- V3 Packaging Tool: ✅ Pass
- ARM-TTK 0.26: 49/50 (known false positive)
- API Versions: Latest stable (2025-09-01 for SecurityInsights)

This PR supersedes and replaces PR #13270 (closed due to branch deletion).

mazamizo21 commented:
## Update: Validation Files Added

Hi Microsoft Team,

We've identified and resolved the CI/CD validation failures:

### Issues Fixed

**1. KqlValidations Failure** ✅
- **Root Cause**: The `Cyren_Indicators_CL` custom table schema was missing from the validation tests
- **Fix**: Added `.script/tests/KqlvalidationsTests/CustomTables/Cyren_Indicators_CL.json` with all 19 table columns
- **Also Added**: `CyrenThreatIntel` to `ValidConnectorIds.json`

**2. WorkbooksValidations Failure** ✅
- **Root Cause**: Preview images were missing for the workbook metadata
- **Fix**: Added preview images (white and black backgrounds) to `Workbooks/Images/Preview/`
- **Updated**: `WorkbooksMetadata.json` with preview image references

### Files Added/Modified

```
+ .script/tests/KqlvalidationsTests/CustomTables/Cyren_Indicators_CL.json
+ Workbooks/Images/Preview/CyrenThreatIntelligenceDashboardWhite.png
+ Workbooks/Images/Preview/CyrenThreatIntelligenceDashboardBlack.png
M .script/tests/detectionTemplateSchemaValidation/ValidConnectorIds.json
M Workbooks/WorkbooksMetadata.json
```

### Expected Results

After this push, the CI/CD checks should now pass:
- ✅ **KqlValidations**: Will pass (table schema and connector ID now registered)
- ✅ **WorkbooksValidations**: Will pass (preview images now present)
- ✅ **All other checks**: Already passing (34/36)

Thank you for your patience!

**Data443 Risk Mitigation, Inc.**

v-maheshbh commented:
Hi @mazamizo21 
Please remove the packageMetadata.json file from the package folder. If these files are required, kindly place them outside the package folder. Also, please update the WorkbooksMetadata file [https://github.com/Azure/Azure-Sentinel/blob/master/Workbooks/WorkbooksMetadata.json] and add the preview images to the Images folder.
You may refer to the below solution from our repository for guidance.
https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/Cloudflare/Workbooks
and
Kindly attach the screenshot of the CCF connector in the connected state for reference.

Thanks!

mazamizo21 commented:
## Update: All Requested Changes Applied

Hi Microsoft Team,

Thank you for your feedback. We have addressed all the requested changes:

### ✅ 1. Removed packageMetadata.json from Package folder
- **Before**: `Solutions/CyrenThreatIntelligence/Package/packageMetadata.json`
- **After**: `Solutions/CyrenThreatIntelligence/packageMetadata.json` (moved to solution root)

### ✅ 2. Updated WorkbooksMetadata.json
- Added Cyren workbook entry with:
  - `workbookKey`: CyrenThreatIntelligenceDashboard
  - `logoFileName`: cyren_logo.svg
  - `dataTypesDependencies`: Cyren_Indicators_CL
  - `dataConnectorsDependencies`: CyrenThreatIntel
  - `previewImagesFileNames`: CyrenThreatIntelligenceDashboardWhite.png, CyrenThreatIntelligenceDashboardBlack.png

### ✅ 3. Added Preview Images
- `Workbooks/Images/Preview/CyrenThreatIntelligenceDashboardWhite.png`
- `Workbooks/Images/Preview/CyrenThreatIntelligenceDashboardBlack.png`

### ✅ 4. CCF Connector Screenshot (Connected State)
- Added screenshot showing connector in Connected state:
  - `Solutions/CyrenThreatIntelligence/Data Connectors/Cyren_CCF/Images/CyrenConnectorConnected.png`

All changes follow the structure of existing solutions (e.g., Cloudflare, Lookout) as referenced.

Thank you!

**Data443 Risk Mitigation, Inc.**

v-maheshbh commented:
Hi @mazamizo21 

Workbook preview images are missing in the path mentioned below. Kindly add the required preview images to the specified directory.
https://github.com/Azure/Azure-Sentinel/tree/master/Workbooks/Images/Preview

Thanks!

mazamizo21 commented:
Investigation completed. I identified the root cause and have applied a fix.

Root Cause:
The workbook preview images existed in the correct location (Workbooks/Images/Preview/) but had a critical format issue. The files were named with .png extensions but were actually JPEG image data. Microsoft's validation system correctly rejected them because they expected actual PNG files.

Verification performed:
- Confirmed both images exist in PR: CyrenThreatIntelligenceDashboardBlack.png and CyrenThreatIntelligenceDashboardWhite.png
- Files were accessible via GitHub raw URL (HTTP 200 responses)
- File format check revealed: JPEG image data despite .png extension

Fix Applied:
Converted both preview images from JPEG to proper PNG format using macOS sips utility:
- CyrenThreatIntelligenceDashboardBlack.png: 437KB (JPEG) to 891KB (PNG)
- CyrenThreatIntelligenceDashboardWhite.png: 451KB (JPEG) to 825KB (PNG)

The corrected PNG files have been committed and pushed to the PR branch. The images now have proper PNG headers and should pass validation.

mazamizo21 commented:
## Update: Deployment Issue Fixed

Hi @v-maheshbh,

Thank you for reporting the deployment issue. We identified and fixed the root cause:

### Root Cause
The `mainTemplate.json` had empty values for workbook template variables:
```json
"workbookVersion1": "",
"workbookContentId1": "",
```

This caused the ARM deployment to fail with `"properties.contentId is required"` error on the workbook contentTemplate resource.

### Fix Applied (Commit `77e03a9875`)
Updated the variables with proper values:
```json
"workbookVersion1": "1.0.0",
"workbookContentId1": "CyrenThreatIntelligenceDashboardWorkbook",
```

### Verification
We verified end-to-end deployment of the Cyren threat intelligence infrastructure:
- ✅ Data Collection Endpoint deployed
- ✅ Data Collection Rules (IP reputation + Malware URLs) deployed
- ✅ Custom table `Cyren_Indicators_CL` created
- ✅ CCF RestApiPoller connectors active and configured
- ✅ Workbooks deployed
- ✅ Cyren API returning valid threat intelligence data

The connectors are using the correct API endpoint: `https://api-feeds.cyren.com/v1/feed/data?feedId=ip_reputation` (and `feedId=malware_urls`).

### Note on Content Hub
The solution uses the V3 Content Hub structure. The CCF connectors and associated infrastructure will only be fully activated after the Content Hub approval is complete. The "Configure Data Connector" flow will become available to users once the solution is approved and published.

Please let us know if you need any additional information.

**Data443 Risk Mitigation, Inc.**

mazamizo21 commented:
## Cyren Threat Intelligence Connector - Connected State

Screenshot showing the CCF connector successfully connected and ingesting threat intelligence data:

✅ Connector Status: Connected
✅ Data Ingestion: Active  
✅ Custom Table: Cyren_Indicators_CL

See attached screenshot: `Cyren.png`

mazamizo21 commented:
<img width="1120" height="928" alt="Cyren" src="https://github.com/user-attachments/assets/d81d3839-0e33-4191-9896-1b83f3f50d99" />

v-maheshbh commented:
Hi @mazamizo21 
Please add the workbook testing document to ensure completeness and proper validation.

Thanks!

mazamizo21 commented:
<img width="1024" height="1024" alt="CyrenThreatIntelligenceDashboardBlack" src="https://github.com/user-attachments/assets/d1f52fd0-7dd7-4ba4-8ad1-d5fdf861048f" />
